js/reference.js

```diff
@@ -1175,42 +1175,59 @@ var ERMrest = (function(module) {
                     // Render markdown content for the pattern
                     this._displayname = module._formatUtils.printMarkdown(pattern, { inline: true });
                 }
-                // no row_name annotation, use column with title, name, or id:text type
-                else if (typeof this._data.title === 'string') {
-                    col = table.columns.get("title");
-                    this._displayname = col.formatvalue(this._data.title, { context: self._pageRef.context });
-                }
-                else if (typeof this._data.name === 'string') {
-                    col = table.columns.get("name");
-                    this._displayname = col.formatvalue(this._data.name, { context: self._pageRef.context });
-                }
-                else  {  
+                // no row_name annotation, use column with title, name, term, label or id:text type
+                // or use the unique key 
+                else {
+
+                    var setDisplaynameForACol = function(name) {
+                        if (typeof self._data[name] === 'string') {
+                            col = table.columns.get(name);
+                            self._displayname = col.formatvalue(self._data[name], { context: self._pageRef.context });
+                            return true;
+                        }
+                        return false;
+                    };
+
+                    var columns = ['title', 'Title', 'name', 'Name', 'term', 'Term', 'label', 'Label'];
```

Comment:
I am not sure that we can consider this case-insensitive e.g. if my column is `TITLE` then this won't detect it. This is especially important for the ID column since I have seen `id`, `Id`, and `ID`. 
 
Should we create a tmp_columns that consists of all column names that have been converted to all lower cases and use this set for checking/mapping in the heuristics?